backend/tracim_backend/lib/utils/authentication.py

```diff
@@ -32,41 +46,148 @@
 AUTH_TOKEN_QUERY_PARAMETER = "access_token"
 CLIENT_TOKEN_HEADER = "X-Tracim-ClientToken"
 
+SAML_IDP_DEFAULT_CONFIG = {
+    "default": {
+        "logo_url": "",
+        "displayed_name": "",
+        "attribute_map": {
+            "user_id": "${mail}",
+            "email": "${eduPersonUniqueId}",
+            "username": "${commonName}",
+            "display_name": "${surName} ${givenName}",
+        },
+        "profile_map": {
+            "trusted-users": {"value": "${eduPersonPrimaryAffiliation}", "match": "teacher|member"},
+            "administrators": {"value": "${eduPersonPrimaryAffiliation}", "match": "employee"},
+        },
+    }
+}
+
+
+def format_attribute(pattern: str, attributes: typing.Dict) -> str:
+    # NOTE - M.L - 2023-10-25 - This regex parses the ${xxx} formatted attribute mapping
+    #  it permits to find and replace multiple patterns in a same string
+    #  \$ Matches the '$' character
+    #  {(.*?)} matches this pattern {xxx} and captures and isolate in a group
+    #  anything between the "{}", it matches between zero and unlimited times
+    #  as few times as possible, meaning that if it meets a '}', it will not be matched
+    placeholders = re.findall(r"\${(.*?)}", pattern)
+    formatted_value = pattern
+    for placeholder in placeholders:
+        if placeholder in attributes:
+            formatted_value = formatted_value.replace(
+                f"${{{placeholder}}}", "".join(attributes[placeholder])
+            )
+    return formatted_value
+
```

Comment:
@Millefeuille42 Probably more simple like this ?

```python
template = "Foo:{foo}, Bar:{bar}, Baz:{baz}"
attributes = {"foo": "42", "bar": "42", "baz": "42"}
print(template.format(**attributes))
```

Note : template.format will fail if key is absent from `attributes` (`KeyError: 'baz'`) and will not fail if `attributes` contain an extra key.